tests/Unit/NumericalAlgorithms/Interpolation/Python/Test_RegularGridInterpolant.py

```diff
@@ -0,0 +1,135 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import unittest
+from spectre import Spectral
+from spectre import DataStructures
+from spectre import Interpolation
+from numpy.polynomial.legendre import Legendre
+import numpy as np
+
+
+class TestRegularGrid(unittest.TestCase):
+
+    # random polynomial functions of low order for exact interpolation
+    def polynomial_1d(self, x):
+        return np.squeeze(x**2 + x + 1)
+
+    def polynomial_2d(self, coords):
+        x, y = coords
+        return 2 * x**2 + y**2 + y + x + 2
+
+    def polynomial_3d(self, coords):
+        x, y, z = coords
+        return 3 * x**2 + 2 * y**2 + z**2 + 2 * y + z + x + 2
+
+    def generate_gauss_nodes(self, num_points):
+        return Legendre.basis(num_points).roots()
+
+    def generate_gauss_lobatto_nodes(self, num_points):
+        nodes = Legendre.basis(num_points - 1).deriv().roots()
+        return np.concatenate(([-1], nodes, [1]))
+
+    def logical_coordinates(self, num_points, dim, quadrature):
+        """
+        creates a uniform mesh of shape (dim, num_points) with the
+        requested quadrature
+        """
+
+        if quadrature == Spectral.Quadrature.Gauss:
+            nodes = self.generate_gauss_nodes(num_points)
+        elif quadrature == Spectral.Quadrature.GaussLobatto:
+            nodes = self.generate_gauss_lobatto_nodes(num_points)
+        else:
+            raise ValueError(
+                "Only Gauss or GaussLobatto are implemented quadratures")
+
+        mesh = np.meshgrid(*tuple(nodes for _ in range(dim)))
+        return np.stack(mesh, 0).reshape(dim, -1)
+
+    def test_1d(self):
+        for quadrature in [
+                Spectral.Quadrature.Gauss, Spectral.Quadrature.GaussLobatto
+        ]:
+            for num_points in range(3, 10):
+                source_mesh = Spectral.Mesh1D(num_points,
+                                              Spectral.Basis.Legendre,
+                                              quadrature)
+                target_mesh = Spectral.Mesh1D(num_points + 2,
+                                              Spectral.Basis.Legendre,
+                                              quadrature)
+                interpolant = Interpolation.RegularGrid1D(
+                    source_mesh, target_mesh)
+
+                source_coords = self.logical_coordinates(
+                    num_points, 1, quadrature)
+                target_coords = self.logical_coordinates(
+                    num_points + 2, 1, quadrature)
```

Comment:
At this point you can make `logical_coordinates` just take the `mesh`. Better than essentially expanding the mesh data into the function :)

Then you need to do only these two things to merge the 1d, 2d and 3d functions into one:
1. Merge the `polynomial_*d` functions: Name `test_polynomial` or similar and determine the dimension from the input shape.
2. Get the `Mesh*D` class like this: `Mesh = [Mesh1D, Mesh2D, Mesh3D][dim - 1]`

We might be able to add a factory like `Mesh(dim=1, ...)`, though I'm not sure if that's necessary and it doesn't need to happen in this PR. That said, if you have an idea how to do that, feel free to give it a shot :D